advancedlog/legacy_logger.go

```diff
@@ -0,0 +1,56 @@
+package logger
+
+import (
+	logutils "github.com/bitrise-io/go-utils/v2/log"
+)
+
+type legacyLogger struct {
+	debugLogEnabled bool
+	logger          logutils.Logger
+}
+
+func newLegacyLogger(logger logutils.Logger) SimplifiedLogger {
+	return &legacyLogger{
+		debugLogEnabled: false,
+		logger:          logger,
+	}
+}
+
+// EnableDebugLog ...
+func (l *legacyLogger) EnableDebugLog(enabled bool) {
+	l.logger.EnableDebugLog(enabled)
+	l.debugLogEnabled = enabled
+}
+
+// IsDebugLogEnabled ...
+func (l *legacyLogger) IsDebugLogEnabled() bool {
+	return l.debugLogEnabled
+}
+
+// LogMessage ...
+func (l *legacyLogger) LogMessage(producer Producer, level Level, message string) {
+	if !l.debugLogEnabled && level == DebugLevel {
+		return
+	}
+
+	// All the print functions below automatically add a newline to the end. We need to replace the newline with an
+	// empty line not to log multiple newlines.
+	if message == "\n" {
```

Comment:
I'm still not sure if we should do any transformation on the given log message (if we misused log calls in the CLI [adding an extra newline at the end], we should simply remove them).

Reply:
updated in https://github.com/bitrise-io/go-utils/pull/166/commits/4323136f49bc1342725c43a9d59be3ae0ac2b0f0